servant-util/src/Servant/Util/Combinators/Logging.hs

```diff
@@ -188,9 +189,16 @@ instance KnownSymbol s => ApiCanLogArg (Capture s a)
 
 instance ApiCanLogArg (ReqBody ct a)
 
-instance KnownSymbol cs => ApiCanLogArg (QueryParam' mods cs a) where
+instance ( Buildable a
+         , KnownSymbol cs
+         , SBoolI (FoldRequired mods)
+         ) => 
+         ApiCanLogArg (QueryParam' mods cs a) where
     type ApiArgToLog (QueryParam' mods cs a) = a
-    toLogParamInfo _ mparam = maybe noEntry pretty mparam
+    toLogParamInfo _ mparam = foldRequiredArgument (Proxy :: Proxy mods) (\(a :: a) -> pretty a)
```

Comment:
I suspect, you don’t need to compute this using `foldRequiredArgument` at runtime. `mods` is known at compile-time, so it should be possible to choose the right pretty-printing implementation at compile-time.

Reply:
Also, I’m not sure if `\(a :: a) -> pretty a` is left from some debugging, or it is really necessary. If it is necessary, it probably deserves a comment :/.

Reply:
Looks like they in servant internals do it via `foldRequiredArgument`, not via a dedicated typeclass, so probably it's fine here as well :thinking:.

The main reason to pick the printing way at type level would be to win bits of performance, right?

Reply:
Can you please explain how make it in here? Because I am not really understand it

Reply:
It... would involve thinking, I don’t want to do it now :). Let’s merge this as is and maybe open an issue so that I’ll think about it later.